### PR TITLE
Changes based on review of pipeline.

### PR DESCRIPTION
**Pull request by kurtzeborn**
- Splitting audit (test and compliance) into separate job
 - Directly calling sn.exe rather than searching for it
 - Building and packaging in a single step
 - Adding Linux job
@weshaggard @shahabhijeet

### CODE REVIEW

**Comment by kurtzeborn** [comment]
The azure-sdk-for-net.client CI is failing because it won't resolve the scope variables I added (scope-L and scope-W).  I've added them to the pipeline definitions, but it just ignores them.  I'm assuming that it will start resolving them once this is merged.  I can't think of another reason why it wouldn't resolve them.

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -28,38 +24,106 @@ jobs:
...
+             if ($proj.Name.Contains("Microsoft.Azure.KeyVault.Tests.csproj"))
+             {
+                 Write-Host "skipping $proj"
+                 continue
+             }
+
+             if ($proj.Name.Contains(".Tests.") -or $proj.Name.Contains(".TestFramework."))
+             {
+                 Write-Host "dotnet restore $proj"
+                 #dotnet restore $proj
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: You can remove the restore steps.

File: .azure-pipelines/client.yml [hunk 2]
```diff
@@ -28,38 +24,106 @@ jobs:
...
-        displayName: 'Publish Artifacts'
-        inputs:
-          PathtoPublish: 'binaries\packages'
-          ArtifactName: packages
+      - script: |
+         "%ProgramFiles(x86)%\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6.1 Tools\sn.exe" -Vu *,31bf3856ad364e35
+        displayName: 'Clean up unsigned build tool allowance'
+        condition: always()
+
+  - job: 'Audit_Linux'
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Lets call these Test_Windows and Test_Linux instead of Audit.

File: .azure-pipelines/client.yml [hunk 3]
```diff
@@ -28,38 +24,106 @@ jobs:
...
+      - powershell: |
+          $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE="true"
+          $env:DOTNET_CLI_TELEMETRY_OPTOUT="1"
+
+          bash ./tools/linuxScripts/getBuildTools.sh
+
+          foreach ($proj in (dir -r -i *.csproj src/$(scope)))
+          {
+             # Have to skip this one because it depends on Windows Cryptography Next Generation (CNG)
+             if ($proj.Name.Contains("Microsoft.Azure.KeyVault.Tests.csproj"))
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Lets be sure to file an issue to track this test project not working on Linux.

File: .azure-pipelines/client.yml (hunk 2 above)
Comment (kurtzeborn) [reply .azure-pipelines/client.yml]: Absolutely.

File: .azure-pipelines/client.yml (hunk 3 above)
Comment (kurtzeborn) [reply .azure-pipelines/client.yml]: Issue opened: https://github.com/Azure/azure-sdk-for-net/issues/5089

File: .azure-pipelines/client.yml (hunk 1 above)
Comment (kurtzeborn) [reply .azure-pipelines/client.yml]: Yep.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**